lib/sorbet-rails/active_record_rbi_formatter.rb

```diff
@@ -0,0 +1,175 @@
+
+  sig {returns(String)}
+  def generate_active_record_base_rbi
+    puts "-- Generate sigs for ActiveRecord::Base --"
+
+    parlour = T.let(Parlour::RbiGenerator.new, Parlour::RbiGenerator)
+
+    parlour.root.add_comments([
+      'This is an autogenerated file for Rails\' ActiveRecord.',
+      'Please rerun bundle exec rake rails_rbi:active_record to regenerate.'
+    ])
+
+    parlour.root.create_class('ActiveRecord::Base') do |class_rbi|
+      create_finder_methods(class_rbi, 'T.attached_class', true)
+    end
+
+    parlour.rbi
+  end
+
+  sig {returns(String)}
+  def generate_active_record_relation_rbi
+    puts "-- Generate sigs for ActiveRecord::Relation --"
+
+    parlour = T.let(Parlour::RbiGenerator.new, Parlour::RbiGenerator)
+
+    parlour.root.add_comments([
+      'This is an autogenerated file for Rails\' ActiveRecord.',
+      'Please rerun bundle exec rake rails_rbi:active_record to regenerate.'
+    ])
+
+    parlour.root.create_class('ActiveRecord::Relation') do |class_rbi|
+      class_rbi.create_constant(
+        "Elem",
+        value: "type_member(fixed: T.untyped)",
+      )
+
+      create_finder_methods(class_rbi, 'Elem', false)
+
+      class_rbi.create_method(
+        "each",
+        parameters: [
+          Parameter.new("&block", type: "T.proc.params(e: Elem).void")
+        ],
+        return_type: "T::Array[Elem]",
+        implementation: true,
+      )
+      class_rbi.create_method(
+        "flatten",
+        parameters: [ Parameter.new("level", type: "T.nilable(Integer)") ],
+        return_type: "T::Array[Elem]",
+      )
+      # this is an escape hatch when there are conflicts in signatures of Enumerable & ActiveRecord
+      class_rbi.create_method(
+        "to_a",
+        return_type: "T::Array[Elem]",
+      )
+      # TODO use type_parameters(:U) when parlour supports it
+      class_rbi.create_arbitrary(
+        code: <<~RUBY
+          sig do
+            type_parameters(:U).params(
+                blk: T.proc.params(arg0: Elem).returns(T.type_parameter(:U)),
+            )
+            .returns(T::Array[T.type_parameter(:U)])
+          end
+          def map(&blk); end
+        RUBY
+      )
+
+      class_rbi.create_method(
+        "exists?",
+        parameters: [ Parameter.new("conditions", type: "T.untyped", default: "nil") ],
+        return_type: "T::Boolean",
+      )
+      class_rbi.create_method('any?', return_type: "T::Boolean")
+      class_rbi.create_method('empty?', return_type: "T::Boolean")
+      class_rbi.create_method('many?', return_type: "T::Boolean")
+      class_rbi.create_method('none?', return_type: "T::Boolean")
+      class_rbi.create_method('one?', return_type: "T::Boolean")
+    end
+
+    parlour.rbi
+  end
+
+  sig {
+    params(
+      class_rbi: Parlour::RbiGenerator::Namespace,
+      type: String,
+      class_method: T::Boolean,
+    ).void
+  }
+  def create_finder_methods(class_rbi, type, class_method)
```

Comment:
Can you make type & class_method keyword arguments? (esp "class_method"). It'll help with reading & comprehending the callsites